packages/compose/src/hooks/use-resize-observer/index.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * External dependencies
+ */
+import useResizeAware from 'react-resize-aware';
+
+/**
+ * Hook which allows to listen the resize event of any target element when it changes sizes.
+ * _Note: `useResizeObserver` will report `null` until after first render_
+ *
+ * @return {Object} Measurements object with properties `width` and `height`.
```

Comment:
Per prior point, for improved typing, this can be:

```suggestion
 * @return {?Object} Measurements object with properties `width` and `height`.
```

Not sure if the documentation generator handles this well.

Related:

- https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#nullable-undefined-and-void-types
- #18045
- #18838